narwhals/_duckdb/expr.py

```diff
@@ -450,9 +452,13 @@ def fill_null(self: Self, value: Any, strategy: Any, limit: int | None) -> Self:
             msg = "todo"
             raise NotImplementedError(msg)
 
-        return self._from_call(
-            lambda _input: CoalesceOperator(_input, lit(value)), "fill_null"
-        )
+        def func(_input: Expression, value: Any) -> Expression:
+            try:
+                return CoalesceOperator(_input, value)
+            except InvalidInputException:
+                return CoalesceOperator(_input, lit(value))
```

Comment:
Is there a more clever way to check we are looking at a literal?

Reply:
thanks for working on this! can you check what we do for other functions where an input might be an expression? i think we should have some logic for this

Reply:
Ha, actually the mere changing the way how value is passed to _from_call did the job. The try-block is unnecessary :-)